23.0.0.2 staging (#2976)
OpenLiberty/blogs: posts/2050-03-07-23.0.0.2.adoc

Check out link:{url-prefix}/blog/?search=release&search!=beta[previous Open Liberty GA release blog posts].

[#run]

== Run your apps using 23.0.0.2

If you're using link:{url-prefix}/guides/maven-intro.html[Maven], here are the coordinates:

[source,xml]
----
<dependency>
    <groupId>io.openliberty</groupId>
    <artifactId>openliberty-runtime</artifactId>
    <version>23.0.0.2</version>
    <type>zip</type>
</dependency>
----

Or for link:{url-prefix}/guides/gradle-intro.html[Gradle]:

[source,gradle]
----
dependencies {
    libertyRuntime group: 'io.openliberty', name: 'openliberty-runtime', version: '[23.0.0.2,)'
}
----

Or if you're using Docker:

[source]
----
FROM open-liberty
----

Or take a look at our link:{url-prefix}/downloads/[Downloads page].

[link=https://stackoverflow.com/tags/open-liberty]
image::img/blog/blog_btn_stack.svg[Ask a question on Stack Overflow, align="center"]

// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // // 
// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/24124
// Contact/Reviewer: aknguyen7,ReeceNana
// // // // // // // // 
[#db_test]
== Test database connections using Admin Center   

> Looking for a simple way to test your database connections? With this beta release, you can now validate your connections using the Liberty Admin Center feature. The connection test exercises the same code paths as your applications, giving you confidence in your server configuration. The Admin Center connection validation feature is enabled by the same REST APIs featured in the link:{url-prefix}/blog/2019/09/13/testing-database-connections-REST-APIs.html[Testing database connections in Open Liberty apps with REST APIs] blog post.

Reviewer: ```suggestion
Looking for a simple way to test your database connections? With this release, you can now validate your connections using the Liberty Admin Center feature. The connection test exercises the same code paths as your applications, giving you confidence in your server configuration. The Admin Center connection validation feature is enabled by the same REST APIs featured in the link:{url-prefix}/blog/2019/09/13/testing-database-connections-REST-APIs.html[Testing database connections in Open Liberty apps with REST APIs] blog post.
```

        <feature>mpOpenApi-3.0</feature>
    </featureManager>

    <!-- To access this server from a remote client add a host attribute to the following element, e.g. host="*" -->
    <httpEndpoint id="defaultHttpEndpoint" httpPort="9080" httpsPort="9443"/>

    <library id="derby">
      <file name="${server.config.dir}/derby/derby.jar"/>
    </library>

    <dataSource id="DefaultDataSource">
      <jdbcDriver libraryRef="derby"/>
      <!-- Example properties referencing an in-memory Derby Embedded database -->
      <properties.derby.embedded databaseName="memory:defaultdb" createDatabase="create"/>
    </dataSource>

    <authData id="myAuth" user="dbuser" password="dbpass"/>

    <!-- Default SSL configuration enables trust for default certificates from the Java runtime --> 
    <ssl id="defaultSSLConfig" trustDefaultCerts="true"/>

    <remoteFileAccess>
       <writeDir>${server.config.dir}</writeDir>
    </remoteFileAccess>

    <basicRegistry id="basic">
       <user name="admin" password="adminpwd"/>
    </basicRegistry>

    <!-- Assign 'admin' to Administrator -->
    <administrator-role>
        <user>admin</user>
    </administrator-role>

</server>
----

> For this `server.xml` example, you will need to add the Derby jar to the server configuration or use your own database configuration.

Reviewer: ```suggestion
For this `server.xml` example, you will need to add the Derby JAR to the server configuration or use your own database configuration.
```

do we need to point to where you can get the Derby JAR? It's not clear

1. Configure a Liberty server using the example `server.xml` file for guidance and then start the server. After the server is started, you can check the logs to find the URL to navigate to the Admin Center. In the previous example, you can navigate to the Admin Center using the `https://localhost:9443/adminCenter/` URL.

> 2. Select the **Server Config** tool.

Reviewer: ```suggestion
2. In the Admin Center UI, select the **Server Config** tool.
```

+
[.img_border_light]
image::img/blog/ac_db_test_server_config.png[Server Config Tool,width=20%,align="center"]

3. Select **server.xml** to edit.

+
[.img_border_light]
image::img/blog/ac_db_test_serverxml.png[server.xml,width=50%,align="center"]

4. In the **Design > Server** menu, navigate to the resource you want to test and click the **Test** button.

+
[.img_border_light]
image::img/blog/ac_db_test_resource.png[Select resource,width=50%,align="center"]

5. Choose the type of authentication your application uses:

+
* For applications that use container authentication, choose the **Application authentication** tab and select whether to use default authentication, specify an authentication alias, or choose a login module configuration.

+

> For this example, the configuration doesn't specify default authentication on the `dataSource` element or configure any login modules. Therefore, you must specify an authentication alias by using the drop-down field.

Reviewer: this line is showing up very small in the draft preview- I thought we fixed that?

Reviewer 2: opened https://github.com/OpenLiberty/openliberty.io/issues/3076

+
[.img_border_light]
image::img/blog/ac_db_test_container_auth.png[Container authentication,width=50%,align="center"]

* For applications that use application authentication, choose the **Application authentication** tab and fill in a valid username and password for the database resource.

+
[.img_border_light]
image::img/blog/ac_db_test_app_auth.png[Application authentication,width=50%,align="center"]

> * If your application does not use a resource reference and the `server.xml` doesn't include `enableContainerAuthForDirectLookups="true"` in the config element, then choose the **No resource reference** tab and fill in a valid username and password for the database resource.

Reviewer: > ...and the `server.xml` doesn't include `enableContainerAuthForDirectLookups="true"` in the config element

which config element?

Reviewer: I tried googling for an answer and it's difficult to find where this attribute is documented. Search results come up for OIDC provider element but I couldn't find the attribute there

Reviewer 3: the enableContainerAuthForDirectLookups is under connectionManager config element https://github.com/OpenLiberty/open-liberty/issues/16230  and https://www.ibm.com/docs/en/was-liberty/base?topic=SSEQTP_liberty/com.ibm.websphere.liberty.autogen.nd.doc/ae/rwlp_config_connectionManager.htm

Reviewer 2: clarified the config element and added link to doc

+
[.img_border_light]
image::img/blog/ac_db_test_no_resource_ref.png[No Resource Reference,width=50%,align="center"]

6. Click the **Connection Test** button to run the test and display the results.  The following example shows a successful connection test:

[.img_border_light]
image::img/blog/ac_db_test_successful_test.png[Successful connection test example,width=50%,align="center"]

> In addition to link:{url-prefix}/docs/latest/reference/feature/jdbc-4.3.html[Java Database Connectivity] you can also test connections to link:{url-prefix}/docs/latest/reference/feature/connectors-2.0.html[Jakarta Connectors], link:{url-prefix}/docs/latest/reference/feature/messaging-3.0.html[Jakarta Messaging] and link:{url-prefix}/docs/latest/reference/feature/cloudant-1.0.html[Cloudant Integration] resources. 

Reviewer: ```suggestion
In addition to link:{url-prefix}/docs/latest/reference/feature/jdbc-4.3.html[Java Database Connectivity], you can also test connections to link:{url-prefix}/docs/latest/reference/feature/connectors-2.0.html[Jakarta Connectors], link:{url-prefix}/docs/latest/reference/feature/messaging-3.0.html[Jakarta Messaging] and link:{url-prefix}/docs/latest/reference/feature/cloudant-1.0.html[Cloudant Integration] resources. 
```

+
[source]
----
.ws.security.openidconnect.clients.common.AttributeToSubject E CWWKS1738E: The OpenID Connect client [client01] failed to authenticate the JSON Web Token because the claim [someBadName] specified by the [userIdentifier] configuration attribute was not included in the token.
----
+
The error message refers to the `userIdentifier` configuration attribute. However, in the `socialLogin-1.0` feature the equivalent configuration attribute is actually called `userNameAttribute`. 
+
This issue has been resolved by updating the NLS message to reference the correct attribute name.

[#guides]
== New and updated guides since the previous release
As Open Liberty features and functionality continue to grow, we continue to add link:https://openliberty.io/guides/?search=new&key=tag[new guides to openliberty.io] on those topics to make their adoption as easy as possible.  Existing guides also receive updates to address any reported bugs/issues, keep their content current, and expand what their topic covers.

* link:{url-prefix}/guides/grpc-intro.html[Streaming messages between client and server services using gRPC] 

> ** Added a cloud-hosted version of this recently published guide.

Reviewer: ```suggestion
** A cloud-hosted version of this recently published guide is now available.
```
